sykefi/marinedataportal
# feat: staggered rendering of sites on map

Currently our site rendering works as follows:
1. get site Ids matching our query
2. split the ids into batches of 200
3. loop through the batches to get all results
4. return all sites at once

This PR changes steps 3 and 4 so that after each batch is returned separately (using a generator function). This means that we start rendering and populating the map as soon as the first batch of 200 sites is received, improving UI responsivity.


# Testing
1. choose secchi depth
2. set time range from the beginning of 2023
3. fetch sites - should render the map quicker and update the sites as long as there are results to fetch
